src/options.js

```diff
@@ -9,101 +9,74 @@ const flashMessage = message => {
-    document.getElementById('wrap-navigation').checked = values.wrapNavigation;
-    document.getElementById('auto-select-first').checked =
-      values.autoSelectFirst;
-    document.getElementById('next-key').value = values.nextKey;
-    document.getElementById('previous-key').value = values.previousKey;
-    document.getElementById('navigate-previous-result-page').value =
-      values.navigatePreviousResultPage;
-    document.getElementById('navigate-next-result-page').value =
-      values.navigateNextResultPage;
-    document.getElementById('navigate-key').value = values.navigateKey;
-    document.getElementById('navigate-new-tab-key').value =
-      values.navigateNewTabKey;
-    document.getElementById('navigate-new-tab-background-key').value =
-      values.navigateNewTabBackgroundKey;
-    document.getElementById('navigate-search-tab').value =
-      values.navigateSearchTab;
-    document.getElementById('navigate-images-tab').value =
-      values.navigateImagesTab;
-    document.getElementById('navigate-videos-tab').value =
-      values.navigateVideosTab;
-    document.getElementById('navigate-maps-tab').value = values.navigateMapsTab;
-    document.getElementById('navigate-news-tab').value = values.navigateNewsTab;
-    document.getElementById('navigate-shopping-tab').value =
-      values.navigateShoppingTab;
-    document.getElementById('navigate-books-tab').value =
-      values.navigateBooksTab;
-    document.getElementById('navigate-flights-tab').value =
-      values.navigateFlightsTab;
-    document.getElementById('navigate-financial-tab').value =
-      values.navigateFinancialTab;
-    document.getElementById('focus-search-input').value =
-      values.focusSearchInput;
-    document.getElementById('navigate-show-all').value = values.navigateShowAll;
-    document.getElementById('navigate-show-hour').value =
-      values.navigateShowHour;
-    document.getElementById('navigate-show-day').value = values.navigateShowDay;
-    document.getElementById('navigate-show-week').value =
-      values.navigateShowWeek;
-    document.getElementById('navigate-show-month').value =
-      values.navigateShowMonth;
-    document.getElementById('navigate-show-year').value =
-      values.navigateShowYear;
-    document.getElementById('toggle-sort').value = values.toggleSort;
+
+    // Handle checks separately.
+    document.getElementById('wrap-navigation').checked = extension.options.sync.values.wrapNavigation;
+    document.getElementById('auto-select-first').checked = extension.options.sync.values.autoSelectFirst;
+    document.getElementById('hide-outline').checked = extension.options.sync.values.hideOutline;
+
+    // Restore options from divs. 
+    for(let key in divToOptionName) {
+
+      // Options are stored as arrays.
+      // Split them into comma-separated string for the uset.
+      document.getElementById(divToOptionName[key]).value = extension.options.sync.values[key].join(', ')
```

Comment:
If a user customized their keybindings, `extension.options.sync.values[key]` will return a string, and the `join` call will fail. That's what I was referring to in the comment about string/array backwards compatibility.